syntax = "proto3";

package structs;

message CassandraInternalField {
  string name = 1;
  string cassandra_type = 2;
}

message CassandraInternalMessage {
  string table_name = 1;

  repeated CassandraInternalField fields = 2;
  repeated CassandraInternalField id_fields = 3;

  string pkg = 4;

  string compaction_strategy = 5;
  string compression_strategy = 6;

  repeated string ids = 7;

  string java_package = 8;
  string java_class = 9;
}

message CassandraInternalMessages {
  repeated CassandraInternalMessage messages = 1;
}
